code/gui/mainwindow.cpp

```diff
@@ -15,17 +17,23 @@ MainWindow::MainWindow(QWidget *parent, const char *title) :
     m_controller_type = Controller::Type::ACTUATOR;
     m_script_engine = std::shared_ptr<ScriptEngine>(new ScriptEngine(m_controller));
 
+    // Bind controller to Python Engine
+    EmbeddedController::getInstance().bind_controller(&m_controller);
+    PythonEngine::getInstance().append_module("emb", &Embedded::PyInit_emb);
+    PythonEngine::getInstance().initialize();
+
     // Setup subwindows
     actuator_setup_window = new ActuatorSetup(m_actuator, this);
     simulator_window = new SimulatorWindow(m_simulator, this);
+    python_interpreter_window = new PythonInterpreter(this);

```

Comment:
looks like you went ahead and created a new class for the interpreter? why not use `ScriptWindow` and the interface I put in place?